compiler/optimizer/LocalOpts.cpp

```diff
@@ -6811,84 +6819,65 @@ int32_t TR_InvariantArgumentPreexistence::perform()
             int32_t index = symbol->getParmSymbol()->getOrdinal();
             _parmInfo[index].setNotInvariant();
             --numInvariantArgs;
-            if (trace())
-               traceMsg(comp(), "PREX:    Arg %d (%s) is not invariant\n", index, node->getSymbolReference()->getName(comp()->getDebug()));
+            traceIfEnabled("PREX:    Arg %d (%s) is not invariant\n", index, node->getSymbolReference()->getName(comp()->getDebug()));
             }
          }
       }
 
-
+   TR::KnownObjectTable *knot = comp()->getKnownObjectTable();
    ListIterator<TR::ParameterSymbol> parms(&methodSymbol->getParameterList());
-   if (_isOutermostMethod)
-      {
-      if (trace())
-         traceMsg(comp(), "PREX:    Populating parmInfo of outermost method %s\n", feMethod->signature(trMemory()));
-
-      int32_t index = 0;
-      TR::ParameterSymbol *p;
-      p = parms.getFirst();
-      if (p && p->getOffset() == 0 && !feMethod->isStatic())
-         {
-         int32_t len = 0;
-         const char *sig = p->getTypeSignature(len);
-         TR_OpaqueClassBlock *clazz = fe()->getClassFromSignature(sig, len, feMethod);
-         _parmInfo[index].setSymbol(p);
-         if (clazz && !fe()->classHasBeenExtended(clazz))
-            {
-            _parmInfo[index].setClassIsCurrentlyFinal();
-            if (trace())
-               traceMsg(comp(), "PREX:      Receiver class is currently final\n");
-            }
-         _parmInfo[index].setClass(clazz);
-         if (trace())
-            traceMsg(comp(), "PREX:      Receiver class %p is %.*s\n", clazz, len, sig);
+   int32_t index = 0;
```

Comment:
I worry the extended scope of index across these two loops may either a) confuse the compiler or b) potentially lead to bugs. If you can't move the index into the loop init could you at least surround each use of index in a `{}` block so so taht the index variables in the two loops are distinct with distinct lifetimes?

Reply:
Updated the code to get index from symbol.